app/js/RateLimitMiddleware.js

```diff
@@ -0,0 +1,16 @@
+const expressRateLimit = require('express-rate-limit')
+const logger = require('logger-sharelatex')
+
+module.exports = {
+  rateLimit
+}
+
+function rateLimit(params) {
+  return expressRateLimit({
+    max: params.max,
+    windowMs: params.windowMs,
+    onLimitReached: req => {
+      logger.warn({ ip: req.ip, params }, 'rate limit reached')
+    }
```

Comment:
Maybe set `headers` to false ([express-rate-limit#headers](https://github.com/nfriedly/express-rate-limit#headers)) so as not to disclose information unnecessarily.